hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/client/ScmClient.java

```diff
@@ -98,12 +98,13 @@ void deleteContainer(long containerId, Pipeline pipeline, boolean force)
    *
    * @param startContainerID start containerID.
    * @param count count must be {@literal >} 0.
+   * @param state Container of this state will be returned.
    *
    * @return a list of pipeline.
    * @throws IOException
    */
   List<ContainerInfo> listContainer(long startContainerID,
```

Comment:
Can we add a new listContainer API with state and keep the existing one for better backward compatibility?

Reply:
This also helps avoiding some unnecessary unit test changes.

Reply:
Sure.  I will add a new listContainer API.  Actually that's the way of my first patch.  Then I thought a consolidated API looks like more neat.